We use LLVM_BUILD_LLVM_DYLIB to build the llvm libraries as shared, but that
does not work for clang. We can use BUILD_SHARED_LIBS on clang, but since we
build them all together, we need to smuggle it in using this patch.

Index: llvm-6.0.0rc1.src/cfe-6.0.0rc1.src/CMakeLists.txt
===================================================================
--- llvm-6.0.0rc1.src.orig/cfe-6.0.0rc1.src/CMakeLists.txt
+++ llvm-6.0.0rc1.src/cfe-6.0.0rc1.src/CMakeLists.txt
@@ -1,5 +1,10 @@
 cmake_minimum_required(VERSION 3.4.3)
 
+if( CLANG_BUILD_SHARED_LIBS )
+  set(BUILD_SHARED_LIBS ON)
+  set(LLVM_BUILD_LLVM_DYLIB OFF)
+endif()
+
 if(POLICY CMP0075)
   cmake_policy(SET CMP0075 NEW)
 endif()
